fedimint-client/src/db.rs

```diff
@@ -262,3 +276,155 @@ impl_db_lookup!(
+                info!(target: LOG_DB, "Migrating module {kind} from {current_db_version} to {target_db_version}");
+                let active_states = get_active_states(&mut global_dbtx.to_ref_nc()).await;
+                let inactive_states = get_inactive_states(&mut global_dbtx.to_ref_nc()).await;
+
+                let new_states = migration(
+                    &mut global_dbtx
+                        .to_ref_with_prefix_module_id(module_instance_id)
+                        .into_nc(),
+                    module_instance_id,
+                    active_states,
+                    inactive_states,
+                )
+                .await?;
+
+                new_states
+            } else {
+                warn!("Missing migration for version {current_db_version}");
+                None
+            };
+
+            if let Some((new_active_states, new_inactive_states)) = new_states {
+                let new_active_states = new_active_states
+                    .into_iter()
+                    .map(|state| (ActiveStateKey::from_state(state), ActiveState::new()))
+                    .collect::<Vec<_>>();
+                let new_inactive_states = new_inactive_states
+                    .into_iter()
+                    .map(|state| {
+                        (
+                            InactiveStateKey::from_state(state),
+                            InactiveState {
+                                created_at: fedimint_core::time::now(),
+                                exited_at: fedimint_core::time::now(),
+                            },
+                        )
+                    })
+                    .collect::<Vec<_>>();
+
+                persist_active_states(&mut global_dbtx.to_ref_nc(), new_active_states).await;
+                persist_inactive_states(&mut global_dbtx.to_ref_nc(), new_inactive_states).await;
+            }
+
+            current_db_version.increment();
+            global_dbtx
+                .insert_entry(&DatabaseVersionKey(module_instance_id), &current_db_version)
+                .await;
+        }
+
+        current_db_version
+    } else {
+        target_db_version
+    };
+
+    global_dbtx.commit_tx_result().await?;
+    info!(target: LOG_DB, "{} module db version: {}", kind, db_version);
+    Ok(())
+}
+
+pub async fn get_active_states(dbtx: &mut DatabaseTransaction<'_>) -> Vec<DynState> {
+    dbtx.find_by_prefix(&ActiveStateKeyPrefix::new())
+        .await
+        .map(|(state, _)| state.state)
+        .collect::<Vec<_>>()
+        .await
+}
+
+pub async fn get_inactive_states(dbtx: &mut DatabaseTransaction<'_>) -> Vec<DynState> {
+    dbtx.find_by_prefix(&InactiveStateKeyPrefix::new())
+        .await
+        .map(|(state, _)| state.state)
+        .collect::<Vec<_>>()
+        .await
+}
+
+pub async fn persist_active_states(
+    dbtx: &mut DatabaseTransaction<'_>,
+    new_active_states: Vec<(ActiveStateKey, ActiveState)>,
```

Comment:
This should probably just take the vector as `Vec<DynState>` and pass the `module_instance_id` as a parameter and do the mapping inside this function